pkg/clients/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,315 @@
+
+	patch, err := CreatePatch(&t, &p)
+	if err != nil {
+		return false, err
+	}
+	return cmp.Equal(&v1alpha1.DynamoTableParameters{}, patch), nil
+}
+
+// IsErrorNotFound helper function to test for ErrCodeTableNotFoundException error
+func IsErrorNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+	return strings.Contains(err.Error(), dynamodb.ErrCodeResourceNotFoundException)
+}
+
+func buildAlphaKeyElements(keys []dynamodb.KeySchemaElement) []v1alpha1.KeySchemaElement {
+	if len(keys) != 0 {
+		keyElements := make([]v1alpha1.KeySchemaElement, len(keys))
+		for i, val := range keys {
+			keyElements[i] = v1alpha1.KeySchemaElement{
+				AttributeName: aws.StringValue(val.AttributeName),
+				KeyType:       string(val.KeyType),
+			}
+		}
+		return keyElements
+	}
+	return nil
+}
+
+func buildTableKeyElements(keys []v1alpha1.KeySchemaElement) []dynamodb.KeySchemaElement {
+	if len(keys) != 0 {
+		keyElements := make([]dynamodb.KeySchemaElement, len(keys))
+		for i, val := range keys {
+			keyElements[i] = dynamodb.KeySchemaElement{
+				AttributeName: aws.String(val.AttributeName),
+				KeyType:       dynamodb.KeyType(val.KeyType),
+			}
+		}
+		return keyElements
+	}
+	return nil
+}
+
+func buildAttributeDefinitions(attributes []dynamodb.AttributeDefinition) []v1alpha1.AttributeDefinition {
+	if len(attributes) != 0 {
+		attributeDefinitions := make([]v1alpha1.AttributeDefinition, len(attributes))
+		for i, val := range attributes {
+			attributeDefinitions[i] = v1alpha1.AttributeDefinition{
+				AttributeName: *val.AttributeName,
+				AttributeType: string(val.AttributeType),
+			}
+		}
+		return attributeDefinitions
+	}
+	return nil
+}
+
+func buildGlobalIndexes(indexes []dynamodb.GlobalSecondaryIndexDescription) []v1alpha1.GlobalSecondaryIndex {
+	if len(indexes) != 0 {
+		globalSecondaryIndexes := make([]v1alpha1.GlobalSecondaryIndex, len(indexes))
+		for i, val := range indexes {
+			globalSecondaryIndexes[i] = v1alpha1.GlobalSecondaryIndex{
+				IndexName: val.IndexName,
+				KeySchema: buildAlphaKeyElements(val.KeySchema),
+				Projection: &v1alpha1.Projection{
+					NonKeyAttributes: val.Projection.NonKeyAttributes,
+					ProjectionType:   string(val.Projection.ProjectionType),
+				},
+			}
+		}
+		return globalSecondaryIndexes
+	}
+	return nil
+}
+
+func buildLocalIndexes(indexes []dynamodb.LocalSecondaryIndexDescription) []v1alpha1.LocalSecondaryIndex {
+	if len(indexes) != 0 {
+		localSecondaryIndexes := make([]v1alpha1.LocalSecondaryIndex, len(indexes))
+		for i, val := range indexes {
+			localSecondaryIndexes[i] = v1alpha1.LocalSecondaryIndex{
+				IndexName: val.IndexName,
+				KeySchema: buildAlphaKeyElements(val.KeySchema),
+				Projection: &v1alpha1.Projection{
+					NonKeyAttributes: val.Projection.NonKeyAttributes,
+					ProjectionType:   string(val.Projection.ProjectionType),
+				},
+			}
+		}
+		return localSecondaryIndexes
+	}
+	return nil
+}
```

Comment:
Nitpick for all of these these `build<T>` functions - it would be nice to 'return early' to move the happy path out of the nested block:

```go
if len(indexes) == 0 {
    return nil
}
localSecondaryIndexes := ...
// etc
return localSecondaryIndexes.
```